### ISSUE TITLE
Allow blank lines in between multiline operator chains?

### ISSUE THREAD

**Issue by rtfeldman** [avh4/elm-format#99]
I spent a bunch of hours trying to find a DSL for elm-css that plays nicely with elm-format.

Here's where I ended up:

```elm
    stylesheet { name = "" }
        $ button
        |- boxSizing borderBox
        |- display inlineBlock

        >$ span
        |- width (pct 100)
        |- height (pct 100)

        $ h1
        |- (or $ h2)
        |- padding zero
        |- margin zero

        >$ h3
        |- width (pct 100)

        >$ h4
        |- height (pct 100)

        $ span
        |- padding (px 10)
        |- margin (px 11)

        >$ ul
        >$ li
        |- width (px 1)
        |- height (pct 2)
```

This *almost* works, except that elm-format removes blank lines in between operator chains, so it currently ends up looking like this:

```elm
    stylesheet { name = "" }
        $ button
        |- boxSizing borderBox
        |- display inlineBlock
        >$ span
        |- width (pct 100)
        |- height (pct 100)
        $ h1
        |- (or $ h2)
        |- padding zero
        |- margin zero
        >$ h3
        |- width (pct 100)
        >$ h4
        |- height (pct 100)
        $ span
        |- padding (px 10)
        |- margin (px 11)
        >$ ul
        >$ li
        |- width (px 1)
        |- height (pct 2)
```

The only barrier to the former is the ability for newlines to be preserved in between multiline operator chains. Would that be a reasonable change to make @avh4?

**Comment by avh4** [comment]
I think it will be reasonable, but I want to finish https://github.com/avh4/elm-format/issues/3 before deciding about this.  The ugliest part is going to be deciding how to deal with comments that may appear at various indentation levels between lines.

You didn't find anything list-based that would work nicely?  (I don't understand what everything is supposed to mean, but here's a stab:)

```elm
    stylesheet { name = "" }
        [ button
            |- boxSizing borderBox
            |- display inlineBlock
        , (>$) span
            |- width (pct 100)
            |- height (pct 100)
        , ($) h1
            |- (or $ h2)
            |- padding zero
            |- margin zero
        , (>$) h3
            |- width (pct 100)
        , (>$) h4
            |- height (pct 100)
        , ($) span
            |- padding (px 10)
            |- margin (px 11)
        , (>$) ul
            >$ li
            |- width (px 1)
            |- height (pct 2)
        ]
```

**Comment by rtfeldman** [comment]
Hot damn, great call @avh4! I never tried it with an outermost list and prefix-style operators.

Assuming I can add blank lines, it looks like I can use that approach to implement a DSL that's really close to my original, except it's [actually elm-format friendly](https://gist.github.com/rtfeldman/b86447e30eb8bacbaf4e)!

**Comment by avh4** [comment]
@rtfeldman can this be closed now in favor of #97 (allowing blank lines in lists)?

**Comment by rtfeldman** [comment]
Yep!
